addons/mail/static/src/components/message_in_reply_to_view/message_in_reply_to_view.xml

```diff
@@ -5,7 +5,14 @@
             <t t-if="messageInReplyToView">
                 <t t-if="!messageInReplyToView.messageView.message.parentMessage.isEmpty">
                     <b class="o_MessageInReplyToView_author text-muted ml-2">@<t t-esc="messageInReplyToView.messageView.message.parentMessage.authorName"/></b>
-                    <span class="o_MessageInReplyToView_body ml-1" t-raw="messageInReplyToView.messageView.message.parentMessage.prettyBody" t-on-click="messageInReplyToView.onClickReply"/>
+                    <t t-if="!messageInReplyToView.messageView.message.parentMessage.isBodyEmpty">
```

Comment:
condition too long on template, I'm suggesting to create a new computed field in messageInReplyToView to hold this value

Reply:
I disagree.

This would artificially make it shorter and give the impression it's simpler, but it'd actually not be the case and instead would introduce a strangely named computed field that is basically alias to something much more meaningful we just did not use this field.

I would like to keep `MessageInReplyToView` being a model that models a message in reply being viewed.
Almost all the time, introduce fields on view records as follow:

- relational fields to logic models that are relevant to this view record
- relational field to parent/children rendering (follows DOM structure)
- attribute fields that are precisely a UI prop/state on this exact view record

Reply:
I don't want to have business logic in template. The decision to display something or not is coming from functional needs.
The template should be dumb and never worry about anything more than "should I display this thing" "yes or no".
We need branching in template to be modeled just like anything else, so that the template can just be a means to get a render done, and not have business on it.

Reply:
> The template should be dumb and never worry about anything more than "should I display this thing" "yes or no".

I get what you mean, and this definitely helps us a lot, in that modelling of state is much better in models (pure modelling in ORM way, much more declarative) than components (very UI-related and tree-shaped architecture, also very algorithmic).

However we don't have yet a declarative way in models to express UI templates, so we still need OWL templates.
If we had this declarative way in models, it would look a bit like current OWL templates... Just not a big algorithmic and more small pieces that can fit together like Legos.

So I get the annoyance of some business logic in component templates, but it's best to have some of them with the current architecture of code. The framework is not ready for no code in templates yet, such as this case.
I would advice not striving too hard for this case, because if you do this when architecture is not ready you will instead implement "solutions" that actually deteriorate the maintainability of code as a whole.

Reply:
It would not be stupid to compute/get/what ever the `parentMessage` inside the `mail.message_in_reply_to_view` model:

- It's the reason the model exist.
- It's almost the only thing we access inside the template.
- It will make the template readable. For a 23 line template it's a lot noisy with long line.

Ultimately, I just want to know what to do. The result is exactly the same from a functional point of vue.

Reply:
I have a feeling you will struggle finding a good and short name for this field.
While the "shape" of the code will be improved, the understanding of the code might be slightly reduced due to cryptic field name and will become a necessary indirection for understanding what this mean.

What gives me these reasons is because this record conceptually models "a message in reply to that is being viewed". View records are just there to mean we refer to this view at hand. They are dumb in nature. The only fields they should have are (1) relational fields to logical records (2) relational fields to view records (view is in context of another view) (3) attribute fields of UI state immediately relevant to this view. By introducing a computed fields that is a logical thing, you just made the view record smart, which goes against view records.

It doesn't cost much to try and see how it goes. Maybe my point of view can be more convincing when you try and see it.

Reply:
In the end, after further discussion, I have changed my mind and it's actually a good idea to introduce computed fields on the view record itself. As long as good concept are introduced in models, it's appropriate to make fields on any records, whether "view" or "pure logic" records.